Nakama/Source/NakamaTests/Private/Tests/Realtime/Test_Match.cpp

```diff
@@ -0,0 +1,300 @@
+			{
+				UE_LOG(LogTemp, Display, TEXT("Added Matchmaker with Ticket: %s"), *Ticket.TicketId);
+				TestTrue("Add Matchmaker Test Passed", !Ticket.TicketId.IsEmpty());
+				StopTest();
+			};
+
+			auto errorCallback = [&](const FNakamaRtError& Error)
+			{
+				UE_LOG(LogTemp, Error, TEXT("Add Matchmaker. ErrorMessage: %s"), *Error.Message);
+				TestFalse("Add Matchmaker Test error.", true);
+				StopTest();
+			};
+
+			// Matchmaker Params
+			// Note: Query: player_level >= 10 does not work. See: https://heroiclabs.com/docs/nakama/concepts/multiplayer/query-syntax/
+			// Doing * instead
+			TOptional<int32> MinCount = 2;
+			TOptional<int32> MaxCount = 4;
+			TOptional<FString> Query = FString("*");
+			TMap<FString, FString> StringProperties;
+			StringProperties.Add("region", "us");
+			TMap<FString, double> NumericProperties;
+			NumericProperties.Add("game_mode", 1);
+			NumericProperties.Add("game_difficulty", 2);
+			TOptional<int32> CountMultiple = 2;
+
+			Socket->AddMatchmaker(MinCount, MaxCount, Query, StringProperties, NumericProperties, CountMultiple, successCallback, errorCallback);
+		});
+
+		Socket->SetDisconnectCallback( [&](const FNakamaDisconnectInfo& DisconnectInfo)
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Socket disconnected: %s"), *DisconnectInfo.Reason);
+			TestFalse("Add Matchmaker Test error, socket disconnected", true);
+			StopTest();
+		});
+		
+		// Connect to Socket
+		Socket->Connect(Session, true);
+	};
+
+	// Define error callback
+	auto errorCallback = [&](const FNakamaError& Error)
+	{
+		// Test fails if there is an authentication error
+		TestFalse("Authentication Test Failed", true);
+		StopTest();
+	};
+
+	// Call the AuthenticateEmail function
+	Client->AuthenticateCustom(FGuid::NewGuid().ToString(), "", true, {}, successCallback, errorCallback);
+
+	// Wait for authentication to complete
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForAsyncQueries(this));
+	
+	// Return true to indicate the test is complete
+	return true;
+}
+
+// Add Matchmaker, wait for Match Matched and Join Match by Token
+IMPLEMENT_CUSTOM_SIMPLE_AUTOMATION_TEST(MatchmakerJoinMatch, FNakamaMatchTestBase, "Nakama.Base.Realtime.Matches"
+																				   ".MatchmakerJoinMatch", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::ProductFilter)
+inline bool MatchmakerJoinMatch::RunTest(const FString& Parameters)
+{
+	// Initiates the test
+	InitiateTest();
+	Client2 = CreateClient();
+	
+	// Define success callback
+	auto successCallback = [&](UNakamaSession* session)
+	{
+		// Set the session for later use
+		Session = session;
+
+		// Setup socket:
+		Socket = Client->SetupRealtimeClient();
+
+		Socket->SetConnectCallback([this]()
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Socket connected"));
+
+			// Add Matchmaker Callbacks
+			auto successCallback = [&](const FNakamaMatchmakerTicket& Ticket)
+			{
+				//UE_LOG(LogTemp, Display, TEXT("Added Matchmaker with Ticket: %s"), *Ticket.TicketId);
```

Comment:
why was this commented out?

Reply:
No particular reason, can be there for clarity